### ISSUE TITLE
Issues with Gatsby

### ISSUE THREAD

**Issue by itzr** [NoahZinsmeister/web3-react#417]
The module `@web3-react/metamask` requires `window`.

This causes build issues with Gatsby.

Gatsby does Server Side Rendering with NodeJs, and therefore no window exists.

It would be great to wrap the web3-react module with:  `const module = typeof window !== 'undefined' ? require("module") : null`, or something similar. I have not tested this solution...It's just an idea inspired by: https://www.gatsbyjs.com/docs/debugging-html-builds/

For now, I have solved the issue by making two changes: 

1. `gatsby-node.js`:

```
exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
    if (stage === "build-html") {
      actions.setWebpackConfig({
        module: {
          rules: [
            {
              test: /@metamask/,
              use: loaders.null(),
            },
            {
                test: /@web3-react/,
                use: loaders.null(),
              },
          ],
        },
      })
    }
  }
```

2. Initializer:
```
export const [metaMask, hooks, store] = typeof window !== `undefined` ? initializeConnector((actions) => new MetaMask(actions)) : [()=>{}, ()=>{},{}]
```
or if you do object destructuring on the hooks, try:
```
export const [metaMask, hooks, store] = typeof window !== `undefined` ? initializeConnector((actions) => new MetaMask(actions)) : [
    ()=>{},
    {
        useChainId: ()=>{},
        useError: ()=>{},
        useIsActive: ()=>{},
        useAccount: ()=>{},
        useIsActivating: ()=>{},
    }   
    ,{}]
```

**Comment by NoahZinsmeister** [comment]
hi @itzr, thanks for flagging! i've recently merged #431, which disallows SSR if the `connectEagerly` flag for any connector is set, as this flag is what kicks off the connection process that in metamask's case assumes the existence of window. so the path forward is to either render exclusively on the client, or to use SSR with `connectEagerly` set to false (while optional calling the new `connectEagerly` for relevant connectors inside a useEffect, to preserve functionality)

the approach you outlined is a bit fragile, as discussed in #427, because it breaks the rules of hooks

let me know if this helps!

**Comment by NoahZinsmeister** [comment]
this should be addressed in the latest `beta` packages, please open another issue if you're still having problems!
